src/lib/core/datetime.c

```diff
@@ -220,6 +221,7 @@ datetime_parse_full(struct datetime *date, const char *str, size_t len,
 	const char *svp = str;
 	char c;
 	int sec_of_day = 0, nanosecond = 0;
+	short tzindex = 0;
```

Comment:
Nit: I'd use `uint16_t`. Especially knowing that it's `uint16_t` in other places.

Reply:
While you mentioned it, I've discovered that there is signed `int16_t` not unsigned `uint16_t` which is not yet adding much value (though eventually might become useful if we would need to return negative values in some contexts, who knows?). 

Changed to `int16_t` - thanks!